### PR TITLE
Add binding paging adapter

### PR DESCRIPTION
**Pull request by Del-S**
- Added `BindingRecyclerViewAdapter`.
- Created `BaseBindingAdapter` to handle basing binding adapter functionality.
- Added `BindingAdapterConfig` for adapter configuration.
- Added builders for adapters.
- Added docs for BindingAdapters.

### CODE REVIEW

File: kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/lifecycle/channels/ChannelExt.kt [hunk 1]
```diff
@@ -68,25 +69,27 @@ fun <A, B> combineLatest(
     var latestB: B? = null
     return scope.produce(context, capacity) {
         whileSelect {
-            channelA.onReceiveOrNull {
+            channelA.onReceiveCatching {
```
Comment (LukasKrystekEman) [inline kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/lifecycle/channels/ChannelExt.kt:72]: Aren't channelA's and channelB's onReceiveCatching implementations the same? Shouldn't it to be written just once?

File: kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BindingAdapters.kt [hunk 2]
```diff
@@ -7,17 +9,19 @@ import androidx.recyclerview.widget.RecyclerView
...
+ * If the adapter already exists it only changes item list in the adapter, which happens with Observable or LiveData
+ * items. If the adapter is a pager adapter it does nothing.
  *
- * @author [eMan a.s.](mailto:info@eman.cz)
+ * @author eMan a.s.
  * @since 0.8.0
  */
 @BindingAdapter(
-    value = ["items", "itemBinder", "variableBinders", "itemOnClick", "itemOnLongClick", "differ", "limit"],
+    value = ["items", "itemBinder", "variableBinders", "itemOnClick", "itemOnLongClick", "differ", "limit", "paging"],
```
Comment (smidf) [inline kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BindingAdapters.kt:24]: Do we want to bind `PagingAdapter`?

File: kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BaseBindingAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    fun getItemViewTypeInternal(position: Int) =
+        getItemInternal(position)?.let { itemBinder.getLayoutRes(it) } ?: 0
+
+    /**
+     * Handles click on the item view by getting the item from view tag and invoking [itemClickListener] if it exists.
+     *
+     * @param view used to get the item and also being send to the click listener
+     */
+    override fun onClick(view: View) {
+        itemClickListener?.let {
```
[the remark this answers is not in the document]
Comment (Del-S) [reply kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BaseBindingAdapter.kt:156]: Fine by me. I would leave it as it is.

File: kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BindingAdapterConfig.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * @property items displayed in the adapter
+ * @property itemBinder defines layout for specific item types
+ * @property variableBinders allows sending custom variables to the layout
+ * @property itemClickListener enables to handle item click (not sub-view click)
+ * @property itemLongClickListener enables to handle item long click (not sub-view click)
+ * @property limit of how many items can be displayed
+ * @author: eMan a.s.
+ * @since 0.9.0
+ */
+data class BindingAdapterConfig<T : Any>(
```
[the remark this answers is not in the document]
Comment (Del-S) [reply kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BindingAdapterConfig.kt]: Visiblity cannot be changed at the moment.

[the remark this answers is not in the document]
Comment (smidf) [reply kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BindingAdapterConfig.kt]: Ok 👍

File: kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BindingAdapters.kt (hunk 2 above)
Comment (smidf) [reply kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/adapter/BindingAdapters.kt:24]: Ok

**Review by smidf — APPROVED** [review]

File: kaal-presentation/build.gradle.kts [hunk 5]
```diff
@@ -52,12 +50,6 @@ android {
     }
 }
 
-tasks.withType<KotlinCompile> {
-    kotlinOptions {
-        jvmTarget = "1.8"
-    }
-}
```
[the remark this answers is not in the document]
Comment (Del-S) [reply kaal-presentation/build.gradle.kts:59]: Sample builds fine and publishToMavenLocal also works. I can try to connect it to some project if you want.

[the remark this answers is not in the document]
Comment (Del-S) [reply kaal-presentation/build.gradle.kts:59]: Well it causes some problems so I returned it in https://github.com/eManPrague/kaal/pull/43/commits/fbb5d81cdbd88830db56bbc25d2e3371f3caa00d.

**Review by smidf — APPROVED** [review]

**Pull request merged**